pinot-controller/src/main/java/org/apache/pinot/controller/validation/UtilizationChecker.java

```diff
@@ -55,4 +56,11 @@ enum CheckPurpose {
     // TASK_GENERATION if the check is performed from the task generation framework to pause creation of new tasks
     REALTIME_INGESTION, TASK_GENERATION
   }
+
+  enum CheckResult {
```

Comment:
Consider using `PASS`, `FAIL` and `UNDETERMINED`? IMO `STALE` is not a property associated with checker, but server stats